File: server/src/main/java/com/mio/server/models/Point.java
package com.mio.server.models;

import com.mio.server.builders.PointBuilder;

public class Point {

    private final Double posX;
    private final Double posY;

    public Point(PointBuilder builder) {
        this.posX = builder.getPosX();
        this.posY = builder.getPosY();
    }

    public Double getPosX() {
        return posX;
    }


    public Double getPosY() {
        return posY;
    }

    @Override
    public String toString() {
        return "Point " +
                "posX " + posX +
                " posY " + posY;
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;

        Point point = (Point) o;

        if (!posX.equals(point.posX)) return false;
        return posY.equals(point.posY);
    }

    @Override
    public int hashCode() {
        int result = posX.hashCode();
        result = 31 * result + posY.hashCode();
        return result;
    }
}

File: Boxworld/app/src/main/java/com/mio/cliente/boxworld/models/Point.kt
package com.mio.cliente.boxworld.models

import com.mio.cliente.boxworld.builders.PointBuilder
import java.io.Serializable

data class Point(val pointBuilder: PointBuilder): Serializable {

    var posX = pointBuilder.posX
    var posY = pointBuilder.posY
    override fun equals(other: Any?): Boolean {
        if (this === other) return true
        if (javaClass != other?.javaClass) return false

        other as Point

        if (posX != other.posX) return false
        if (posY != other.posY) return false

        return true
    }

    override fun hashCode(): Int {
        var result = posX ?: 0
        result = 31 * result + (posY ?: 0)
        return result
    }

    override fun toString(): String {
        return "Point(posX=$posX, posY=$posY)"
    }


}